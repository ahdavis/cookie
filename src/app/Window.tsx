/*
 * Window.tsx
 * Defines the window component for Cookie
 * Created on 11/22/2019
 * Created by Andrew Davis
 *
 * Copyright (C) 2019  Andrew Davis
 *
 * This program is free software: you can redistribute it and/or modify   
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 * 
 * You should have received a copy of the GNU General Public License
 * along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */

//imports
import * as React from 'react';
import { WindowProps } from '../props/WindowProps';
import { WindowState } from '../state/WindowState';
import { Screen } from '../gfx/Screen';

/**
 * The window of the app
 */
export class Window extends React.Component<WindowProps, WindowState> {
	//no fields

	/**
	 * Constructs a new `Window` component
	 *
	 * @param props The properties of the `Window`
	 */
	constructor(props: WindowProps) {
		super(props); //call the superclass constructor
	}

	/**
	 * Renders the `Window`
	 *
	 * @returns A React DOM representing the `Window`
	 */
	public render(): React.ReactNode {
		return (
			<Screen frame={this.props.frame} />
		);
	}
}

//end of file
