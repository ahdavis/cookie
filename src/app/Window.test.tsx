/*
 * Window.test.tsx
 * Defines unit tests for the Window component
 * Created on 11/22/2019
 * Created by Andrew Davis
 *
 * Copyright (C) 2019  Andrew Davis
 *
 * This program is free software: you can redistribute it and/or modify   
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 * 
 * You should have received a copy of the GNU General Public License
 * along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */

//imports
import * as React from 'react';
import * as ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import { Window } from './Window';

//start of tests
describe('Window', () => {
	//this test verifies that the window renders without crashing
	it('Renders without crashing', () => {
	  const div = document.createElement('div');
	  ReactDOM.render(<Window />, div);
	  ReactDOM.unmountComponentAtNode(div);
	});

	//this test verifies that the window matches its snapshot
	it('Renders to snapshot', () => {
		const win = shallow(<Window />);
		expect(win).toMatchSnapshot();
	});
});


//end of tests
