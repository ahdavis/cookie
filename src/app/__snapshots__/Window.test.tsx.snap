// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Window Renders to snapshot 1`] = `ShallowWrapper {}`;
